script/verdaccio.js

```diff
@@ -0,0 +1,220 @@
+  const SDK_APP_NAME = 'amazon-chime-sdk-js';
+  const applicationProcesses = [];
+
+  const cleanupBeforeExit = () => {
+    logger.log('Terminating processes...');
+    applicationProcesses.map(process => {
+      logger.log(`Killing process: ${process.pid}`);
+      spawnOrFail('kill', [`kill -9`, `${process.pid}`]);
+    });
+    spawnOrFail('ps', [
+      `-ef | grep -E '(webpack|verdaccio)' | grep -v grep | awk '{ print $2 }' | xargs kill -9`
+  ]);
+    process.exit(0);
+  };
+  
+  process.on('SIGTERM', () => {
+    logger.warn("Received SIGTERM, cleaning up.")
+    cleanupBeforeExit();
+  });  
+  process.on('SIGINT', () => {
+    logger.warn("Received SIGINT, cleaning up.")
+    cleanupBeforeExit();
+  });
+  process.on('error', (e) => {
+      logger.error(`Error received: ${e}`);
+      cleanupBeforeExit();
+  })
+  
+  // Create verdaccio testing directory and cd in to it
+  if (!fs.existsSync(verdaccioAppDir)) {
+    logger.log(`Creating verdaccio directory: ${verdaccioAppDir}`);
+    fs.mkdirSync(verdaccioAppDir);
+  }
+  
+  //  Go to verdaccio dir
+  logger.log(`Change directory to verdaccio directory: ${verdaccioAppDir}`);
+  process.chdir(verdaccioAppDir);
+
+  // Initiate basic node package
+  logger.log(`Creating basic node package`);
+  spawnOrFail('npm', ['init', '-f', '-y']);
+  
+  // Install verdaccio within new package
+  logger.log(`Installing Verdaccio`);
+  spawnOrFail('npm', ['i', 'verdaccio']);
+  
+  generateVerdaccioConfigFile();
+  
+  // Start verdaccio application
+  try {
+    startVerdaccioApp();
+  } catch (e) {
+    logger.error('Failed to start verdaccio', e);
+  }
+  
+  // Make SDK test directory
+  if (!fs.existsSync(sdkTestDir)) {
+    logger.log(`Creating SDK testing directory: ${sdkTestDir}`);
+    fs.mkdirSync(sdkTestDir);
+  } else {
+    logger.log(`SDK testing directory already exists: ${sdkTestDir}`);
+  }
+
+  logger.log(`Navigate to SDK testing directory: ${sdkTestDir}`);
+  process.chdir(sdkTestDir);
+  
+  // Clone SDK from github
+  logger.log(`Cloning ${SDK_APP_NAME} into ${sdkTestDir}`);
+  
+  //Clean up
+  spawnOrFail('rm', ['-rf', `${SDK_APP_NAME}`]);
+  spawnOrFail('git', [
+    'clone',
+    'git@github.com:aws/amazon-chime-sdk-js.git'
+  ]);
+  // navigate in to cloned SDK
+  process.chdir(path.join(`./${SDK_APP_NAME}`));
+  
+  // Update package.json version
+  let package = fs.readFileSync('package.json', 'utf-8');
+
+  const packageJson = JSON.parse(package);
+  const originalVersion = packageJson.version;
+  const versions = originalVersion.split('.').map(x => parseInt(x));
+  versions[2] += 1;
+  const newVersion = versions.join('.');
+
+  logger.log(`Changing version number from ${originalVersion} to ${newVersion}`);
+
+  // Update version to new version
+  let versionFile = fs.readFileSync('./src/versioning/Versioning.ts', 'utf-8');
+  fs.writeFileSync('./src/versioning/Versioning.ts', versionFile.replace('return VERSION.semverString;', `return '${newVersion};'`));
```

Comment:
We should not need to do that anymore after Richard's change.
To update package version, we may want to use npm version --no-git-tag-version like in publish script.

Reply:
This is following the current steps in the MCM, we manually change this version before we push to verdaccio. Calling 'npm version' would only update hte package.json version, not this folder. I noticed we have a script called generate-version.js, but since this is just a dryrun essentially, I think we can jsut manually update this Versioning file, like we do in the MCM today.

Reply:
We should update the MCM to remove the manually update the versioning. That is not how the publish script works anymore.